managed/services/management/backup/artifacts_service.go

```diff
@@ -104,15 +104,36 @@ func (s *ArtifactsService) ListArtifacts(context.Context, *backuppb.ListArtifact
 	}, nil
 }
 
-// DeleteArtifact deletes specified artifact.
+// DeleteArtifact deletes specified artifact and its files.
 func (s *ArtifactsService) DeleteArtifact(
 	ctx context.Context,
 	req *backuppb.DeleteArtifactRequest,
 ) (*backuppb.DeleteArtifactResponse, error) {
+	if req.RemoveFiles {
+		artifact, err := models.FindArtifactByID(s.db.Querier, req.ArtifactId)
+		if err != nil {
+			return nil, err
+		}
+
+		service, err := models.FindServiceByID(s.db.Querier, artifact.ServiceID)
+		if err != nil {
+			return nil, err
+		}
+
+		if service.ServiceType == models.MongoDBServiceType && artifact.Mode == models.PITR {
+			location, err := models.FindBackupLocationByID(s.db.Querier, artifact.LocationID)
+			if err != nil {
+				return nil, err
+			}
+			if err = s.pbmPITRService.DeletePITRChunks(ctx, location, artifact, nil); err != nil {
+				return nil, err
+			}
+		}
```

Comment:
Let's wrap it into TX

Reply:
For non-serializable tx it will not change anything. Anyway logic was changed.